web/src/pages/eventslist.js

```diff
@@ -0,0 +1,126 @@
+import React, {Component} from 'react';
+
+import Eventspanel from "../components/eventslist-component/eventspanel";
+import Paper from "@material-ui/core/Paper";
+import Tabs from "@material-ui/core/Tabs";
+import Tab from "@material-ui/core/Tab";
+import Typography from "@material-ui/core/Typography";
+import Box from "@material-ui/core/Box";
+
+import PropTypes from 'prop-types';
+import Grid from "@material-ui/core/Grid";
+
+function TabPanel(props) {
+  const { children, value, index, ...other } = props;
+
+  return (
+      <Typography
+          component="div"
+          role="tabpanel"
+          hidden={value !== index}
+          id={`scrollable-auto-tabpanel-${index}`}
+          aria-labelledby={`scrollable-auto-tab-${index}`}
+          {...other}
+      >
+        {/* if value == index, return the children(cards component) for each panel */}
+        {value === index && <Box p={3}>{children}</Box>}
+      </Typography>
+  );
+}
+// define the type of prop of TabPanel
+TabPanel.propTypes = {
+  children: PropTypes.node,
+  index: PropTypes.any.isRequired,
+  value: PropTypes.any.isRequired,
+};
+
+class Eventslist extends Component {
+  constructor(props) {
+    super(props);
+
+    const places = [
```

Comment:
Why are these hard coded here? Shouldn't these be populated from the responses from backend? What happens if the backend doesn't return any responses containing these cities? FE code shouldn't have its own business logic that can be computed from BE.

Reply:
I remove the tab component as area db of event is not designed.